test/connection_string/test/valid-auth.json

```diff
@@ -240,6 +240,27 @@
         "authmechanism": "MONGODB-CR"
       }
     },
+    {
+      "description": "Subdelimiters in user/pass don't need escaping (MONGODB-CR)",
```

Comment:
I am not sure if this test is correctly written but I'm guessing that it is, since a lot of other drivers have already implemented this spec change. The main reason I am skeptical is that the username/password here `!$&'()*+,;=` contains a `+` which, when `unquote`-ed with `unqoted_plus` method drops the `+` in favor of a space, but we expect the parsed username/password to be identical to what is in the input string. Consequently, I had to modfiy the `parse_userinfo` method to use the `unquote` method.

Reply:
Yeah we definitely can't make this change in 3.9. Technically we can do it in 4.0 since major versions allow breaking changes but I'm not sure we should do it ever. If we make this change then users will have to change all their URIs and code that already relies on `quote_plus`. That sounds very painful.

Reply:
I think we need to deviate from the spec a bit for backwards compatibility. We should change the "Username with password containing an unescaped percent sign" to `"valid": true` and change "Subdelimiters in user/pass don't need escaping (MONGODB-CR)" to:
```
        "username": "!$&'()* ,;=",
        "password": "!$&'()* ,;=",
```

This will be annoying for us to maintain because we'll need to reapply these patches every time we update the spec tests. Maybe there's a simpler way?